nixos/maintainers/scripts/ec2/amazon-hvm-userdata-config.nix

```diff
@@ -0,0 +1,35 @@
+{ config, pkgs, ...}:
+
+let
+  bootScript = pkgs.writeScript "bootscript.sh" ''
+    #!${pkgs.stdenv.shell} -eux
+
+    export PATH=${pkgs.nix}/bin:${pkgs.curl}/bin:${pkgs.systemd}/bin:${pkgs.gnugrep}/bin:${pkgs.gnused}/bin:${config.system.build.nixos-rebuild}/bin:$PATH
+    export NIX_PATH=/nix/var/nix/profiles/per-user/root/channels/nixos:nixos-config=/etc/nixos/configuration.nix
+
+    while ! curl -s --max-time 5 169.254.169.254 > /dev/null; do
+      sleep 5
+    done
+
+    echo "Success"
+
+    curl -s http://169.254.169.254/2011-01-01/user-data > /etc/nixos/amazon-init.nix
```

Comment:
I wanted a simple place that would make it clear what came from user-data and what didn't. That way, configuration.nix comes pre-wired to import from amazon-init.nix but if you want to jump in and change it, you can tell it to stop importing amazon-init.nix, or use the existing file differently and so on. It also feels weird to auto-write to a location that people typically treat as human-managed. If someone were generating their own AMI, they wouldn't run the risk of getting their handmade configuration clobbered by my startup script.